source/common/http/http2/codec_impl.cc

```diff
@@ -349,6 +354,39 @@ void ConnectionImpl::StreamImpl::encodeMetadata(const MetadataMapVector& metadat
   }
 }
 
+void ConnectionImpl::StreamImpl::processBufferedData() {
+  ENVOY_CONN_LOG(debug, "Stream {} processing buffered data.", parent_.connection_, stream_id_);
+
+  // We should stop processing buffered data if either
+  // 1) Buffers become overrun
+  // 2) The stream ends up getting reset
```

Comment:
What if sending data results in an error and there's a local reply generated instead? 
Or does downstream local reply reset unfinished upstream stream so we're Ok just using reset?  I think we'll want an integration test for both downstream reset and downstream local reply either way.

Reply:
I think it's the latter, for example in the grpc json transcoder test, if we're above a threshold in the filter we end up sending a local reply as a result of processing this buffered data. Will add some other tests to explicitly test this though.